File: docs/snippet/code-signal/arcade/intro/island-of-knowledge/box-blur/box-blur.py
def box_blur(image):
    result = []
    for i in range(len(image) - 2):
        row = []
        for j in range(len(image[0]) - 2):
            row.append(
                ((sum(image[i][j:j+3])) +
                 (sum(image[i+1][j:j+3])) +
                 (sum(image[i+2][j:j+3]))
                 ) // 9
            )
        result.append(row)
    return result


print(box_blur([
    [36, 0, 18, 9],
    [27, 54, 9, 0],
    [81, 63, 72, 45]
]))

File: content/snippet/code-signal/arcade/intro/island-of-knowledge/box-blur/box-blur.js
function box_blur(image) {
  const result = [];
  for (let i = 0; i < image.length - 2; i++) {
    const row = [];
    for (let j = 0; j < image[i].length - 2; j++) {
      let sum = 0;
      for (let k = i; k < i + 3; k++) {
        for (let l = j; l < j + 3; l++) {
          sum += image[k][l];
        }
      }
      row.push(Math.floor(sum / 9));
    }
    result.push(row);
  }
  return result;
}

console.log(
  box_blur([
    [36, 0, 18, 9],
    [27, 54, 9, 0],
    [81, 63, 72, 45],
  ])
);
